lib/openstack_query/query_wrapper.py

```diff
@@ -0,0 +1,353 @@
+from enum import Enum
+from typing import Tuple, Callable, Any, Dict, Optional, List, Union
+from openstack_api.openstack_wrapper_base import OpenstackWrapperBase
+from openstack_api.openstack_connection import OpenstackConnection
+
+from openstack_query.utils import (
+    check_filter_func,
+    check_kwarg_mapping,
+    prop_equal_to,
+    prop_not_equal_to,
+    prop_greater_than,
+    prop_less_than,
+    prop_younger_than,
+    prop_older_than,
+    prop_older_than_or_equal_to,
+    prop_younger_than_or_equal_to,
+    prop_any_in,
+    prop_not_any_in,
+    prop_matches_regex,
```

Comment:
Ditto about another class for these too. We could further split these down into

`QueryPropDate` # For younger or older...etc.
`QueryPropSets`  # For any in or not in 
`QueryPropComparisons`  # For equal / NE / regex